cypress/support/ui_commands.js

```diff
@@ -135,13 +134,12 @@ Cypress.Commands.add('postMessageReplyInRHS', (message) => {
 });
 
 Cypress.Commands.add('getLastPost', () => {
-    return cy.get('#postListContent #postContent', {timeout: TIMEOUTS.MEDIUM}).last();
+    cy.get('#post-list', {timeout: TIMEOUTS.HUGE}).should('be.visible');
+    return cy.get('#postListContent #postContent', {timeout: TIMEOUTS.HUGE}).last().scrollIntoView().should('be.visible');
```

Comment:
checking visibility of the post shouldn't be here

Reply:
Ok, this one was a bit gnarly. There was a test (and this may have been specific to the data that was in the channel), but it was having issues targeting the last post. It seemed like the posts weren't done loading and it was finding the "last post at that time", which wasn't the actual last post. Having the scrollIntoView() and the check helped me out in this case (again, safety assertion per a previous comment of mine). 

Do you think we should omit the scrollIntoView() as well? The way we have been using it, it feels like the expectation is to be "hey, give me the last post and show it to me", but maybe I'm wrong?

Reply:
> @cometkim I'm on the fence of omitting validation.

I apologize for the ambiguous comments. I'm not against you, I just thought assertions and setup should be separated explicitly.

For example, like this:
```
// setup
if (makePrivate) {
    cy.get('#private').check({force: true}).as('channelType')
} else {
    cy.get('#public').as('channelType')
}

// and validate
cy.get('@channelType').should('be.checked')
```

I usually check the behavior on the community server, so I can not confirm that the last post did not load first. (Sounds like a bug) However, given the purpose of getLastPost, I thought that implicit validation would cause other users to create some duplications.

Reply:
@cometkim I like your approach to this validation way better, will update.

And yeah the post thing is really dependent on the test and if you are hitting the page fresh and trying to get the last post. At this point I'm ok with possible duplicates (which hopefully can be caught during code reviews) to hopefully retain more stability for tests. I'd rather have this command handle some basic level assertions, rather than having other users/community members have to remember to check for these things.